{"uid":"7e9477bf1d676706","name":"Проверка валидации поля ввода кода из смс при вводе неверного значения","fullName":"tests.test_reset_password.TestResetPassword#test_receiving_sms_code_to_reset_password","historyId":"57bcb3085b34ce1268c6f60bfaf61447","time":{"start":1737571400685,"stop":1737571436966,"duration":36281},"description":"Для восстановления используется номер от Aleska","descriptionHtml":"<p>Для восстановления используется номер от Aleska</p>\n","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: Не найден элемент с локатором ('xpath', '//input[@name=\"code\"]')","statusTrace":"self = <test_reset_password.TestResetPassword object at 0x7fc3f13d0260>\nelements = {'Банк ачивок': {'Блок ачивок': ('xpath', '//div[contains(@class,\"user-achievement-by-interests_container\")]'), 'Модал...-sheet_icons\")]'), 'Название ачивки': ('xpath', '//div[contains(@class,\"achievement-item-bottom-sheet_title\")]')}, ...}\n\n    @allure.title('Проверка валидации поля ввода кода из смс при вводе неверного значения')\n    @allure.description('Для восстановления используется номер от Aleska')\n    @allure.severity('Critical')\n    @pytest.mark.regression\n    def test_receiving_sms_code_to_reset_password(self, elements):\n        with allure.step('Открыть страницу логина'):\n            self.login_page.open()\n            element = elements['Страница логина']\n    \n        with allure.step('Нажать кнопку забыл пароль'):\n            self.login_page.do_click(element['Кнопка забыл пароль'])\n    \n        with allure.step('В поле \"Email или номер телефона\" ввести номер телефона от Aleska'):\n            element = elements['Страница восстановления пароля']\n            self.reset_password_page.field_send_keys(element['Поле логин'],\n                                                     text=self.User.PHONE)\n    \n        with allure.step('Нажать кнопку \"Сбросить пароль\"'):\n            self.reset_password_page.do_click(element['Кнопка сбросить пароль'])\n    \n        with allure.step('В поле Код из смс ввести смс код'):\n>           self.reset_password_page.field_send_keys(element['Поле код из смс'],\n                                                     text=f'{fake.pyint(min_value=1000, max_value=9999)}')\n\ntests/test_reset_password.py:131: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nbase/base_page.py:126: in field_send_keys\n    element = self.find_element(locator)\nbase/base_page.py:138: in find_element\n    return self.wait.until(EC.visibility_of_element_located(locator),\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ba13abd586ba869c4f32b100f577b855\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7fc3f15a2840>\nmessage = 'Не найден элемент с локатором (\\'xpath\\', \\'//input[@name=\"code\"]\\')'\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: Не найден элемент с локатором ('xpath', '//input[@name=\"code\"]')\n\n../local/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1737569139111,"stop":1737569139113,"duration":2},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false},{"name":"driver","time":{"start":1737571400203,"stop":1737571400684,"duration":481},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false},{"name":"Инициализация локаторов","time":{"start":1737571400684,"stop":1737571400685,"duration":1},"status":"passed","steps":[{"name":"Инициализация локаторов","time":{"start":1737571400684,"stop":1737571400685,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[{"name":"request","value":"<SubRequest 'elements' for <Function test_receiving_sms_code_to_reset_password>>"}],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":1,"hasContent":true,"attachmentStep":false},{"name":"setup","time":{"start":1737571400684,"stop":1737571400684,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Для восстановления используется номер от Aleska","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: Не найден элемент с локатором ('xpath', '//input[@name=\"code\"]')","statusTrace":"self = <test_reset_password.TestResetPassword object at 0x7fc3f13d0260>\nelements = {'Банк ачивок': {'Блок ачивок': ('xpath', '//div[contains(@class,\"user-achievement-by-interests_container\")]'), 'Модал...-sheet_icons\")]'), 'Название ачивки': ('xpath', '//div[contains(@class,\"achievement-item-bottom-sheet_title\")]')}, ...}\n\n    @allure.title('Проверка валидации поля ввода кода из смс при вводе неверного значения')\n    @allure.description('Для восстановления используется номер от Aleska')\n    @allure.severity('Critical')\n    @pytest.mark.regression\n    def test_receiving_sms_code_to_reset_password(self, elements):\n        with allure.step('Открыть страницу логина'):\n            self.login_page.open()\n            element = elements['Страница логина']\n    \n        with allure.step('Нажать кнопку забыл пароль'):\n            self.login_page.do_click(element['Кнопка забыл пароль'])\n    \n        with allure.step('В поле \"Email или номер телефона\" ввести номер телефона от Aleska'):\n            element = elements['Страница восстановления пароля']\n            self.reset_password_page.field_send_keys(element['Поле логин'],\n                                                     text=self.User.PHONE)\n    \n        with allure.step('Нажать кнопку \"Сбросить пароль\"'):\n            self.reset_password_page.do_click(element['Кнопка сбросить пароль'])\n    \n        with allure.step('В поле Код из смс ввести смс код'):\n>           self.reset_password_page.field_send_keys(element['Поле код из смс'],\n                                                     text=f'{fake.pyint(min_value=1000, max_value=9999)}')\n\ntests/test_reset_password.py:131: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nbase/base_page.py:126: in field_send_keys\n    element = self.find_element(locator)\nbase/base_page.py:138: in find_element\n    return self.wait.until(EC.visibility_of_element_located(locator),\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ba13abd586ba869c4f32b100f577b855\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7fc3f15a2840>\nmessage = 'Не найден элемент с локатором (\\'xpath\\', \\'//input[@name=\"code\"]\\')'\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: Не найден элемент с локатором ('xpath', '//input[@name=\"code\"]')\n\n../local/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","steps":[{"name":"Открыть страницу логина","time":{"start":1737571400685,"stop":1737571403283,"duration":2598},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false},{"name":"Нажать кнопку забыл пароль","time":{"start":1737571403283,"stop":1737571405217,"duration":1934},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false},{"name":"В поле \"Email или номер телефона\" ввести номер телефона от Aleska","time":{"start":1737571405217,"stop":1737571406802,"duration":1585},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false},{"name":"Нажать кнопку \"Сбросить пароль\"","time":{"start":1737571406802,"stop":1737571406843,"duration":41},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":0,"stepsCount":0,"hasContent":false,"attachmentStep":false},{"name":"В поле Код из смс ввести смс код","time":{"start":1737571406843,"stop":1737571436966,"duration":30123},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: Не найден элемент с локатором ('xpath', '//input[@name=\"code\"]')\n\n","statusTrace":"  File \"/usr/workspace/tests/test_reset_password.py\", line 131, in test_receiving_sms_code_to_reset_password\n    self.reset_password_page.field_send_keys(element['Поле код из смс'],\n  File \"/usr/workspace/base/base_page.py\", line 126, in field_send_keys\n    element = self.find_element(locator)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/workspace/base/base_page.py\", line 138, in find_element\n    return self.wait.until(EC.visibility_of_element_located(locator),\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"attachmentsCount":0,"stepsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"attachmentsCount":0,"stepsCount":5,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"driver::0","time":{"start":1737571436978,"stop":1737571437169,"duration":191},"status":"passed","steps":[],"attachments":[{"uid":"29703420f84d2e02","name":"page_source","source":"29703420f84d2e02.html","type":"text/html","size":10500},{"uid":"8d8e3d22646de7d7","name":"screenshot","source":"8d8e3d22646de7d7.png","type":"image/png","size":29906},{"uid":"e2595d463e1d38ee","name":"browser_logs","source":"e2595d463e1d38ee.txt","type":"text/plain","size":1208}],"parameters":[],"shouldDisplayMessage":false,"attachmentsCount":3,"stepsCount":0,"hasContent":true,"attachmentStep":false}],"labels":[{"name":"feature","value":"Восстановление пароля"},{"name":"severity","value":"Critical"},{"name":"tag","value":"regression"},{"name":"tag","value":"reset_password"},{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_reset_password"},{"name":"subSuite","value":"TestResetPassword"},{"name":"host","value":"5230432a8515"},{"name":"thread","value":"1-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_reset_password"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"critical","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":1,"broken":6,"skipped":0,"passed":31,"unknown":0,"total":38},"items":[{"uid":"52196401c91adcf9","status":"broken","statusDetails":"selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited abnormally.\n  (unknown error: DevToolsActivePort file doesn't exist)\n  (The process started from chrome location /usr/bin/chromium-browser is no longer running, so ChromeDriver is assuming that Chrome has crashed.)","time":{"start":1737568973982,"stop":1737568973982,"duration":0}},{"uid":"5b63854eaf99e257","status":"passed","time":{"start":1734035294260,"stop":1734035302607,"duration":8347}},{"uid":"60665cf302993916","status":"passed","time":{"start":1733938101607,"stop":1733938109039,"duration":7432}},{"uid":"ab276b75bbef556f","status":"passed","time":{"start":1733866733660,"stop":1733866741186,"duration":7526}},{"uid":"cfc48d17b3150650","status":"passed","time":{"start":1733861170284,"stop":1733861177991,"duration":7707}},{"uid":"f22758073f34cabb","status":"passed","time":{"start":1733857590599,"stop":1733857598322,"duration":7723}},{"uid":"3e17f61e91aaddf0","status":"passed","time":{"start":1733772563962,"stop":1733772571706,"duration":7744}},{"uid":"bf12d3d0dd6440eb","status":"passed","time":{"start":1733766987188,"stop":1733766995226,"duration":8038}},{"uid":"16e84360b3d9c705","status":"passed","time":{"start":1733422590381,"stop":1733422598770,"duration":8389}},{"uid":"f15fa4e2d1b5ceb8","status":"passed","time":{"start":1726780870182,"stop":1726780877765,"duration":7583}},{"uid":"dcfe6e2a3d7986ab","status":"broken","statusDetails":"selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: element is not attached to the page document\n  (Session info: headless chrome=112.0.5615.165); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception","time":{"start":1726777645039,"stop":1726777649677,"duration":4638}},{"uid":"676fef7a0f8d015d","status":"passed","time":{"start":1724099393327,"stop":1724099402339,"duration":9012}},{"uid":"3e86e9a4e8c1d7c6","status":"passed","time":{"start":1723829735291,"stop":1723829744373,"duration":9082}},{"uid":"1df72a69b2b08636","status":"passed","time":{"start":1723751861075,"stop":1723751869002,"duration":7927}},{"uid":"9a316cc3a1276694","status":"passed","time":{"start":1723742342163,"stop":1723742349929,"duration":7766}},{"uid":"6deaa835e74921a9","status":"passed","time":{"start":1723664749576,"stop":1723664757607,"duration":8031}},{"uid":"3e65d556c32454c4","status":"passed","time":{"start":1723579469924,"stop":1723579478743,"duration":8819}},{"uid":"5bc088e55f7bc38","status":"passed","time":{"start":1723496622575,"stop":1723496630279,"duration":7704}},{"uid":"f1d3f9e561a52758","status":"broken","statusDetails":"selenium.common.exceptions.TimeoutException: Message: Не найден элемент с локатором ('xpath', '//input[@name=\"code\"]')","time":{"start":1723150424060,"stop":1723150458827,"duration":34767}},{"uid":"909951c32664433d","status":"passed","time":{"start":1723149107041,"stop":1723149114121,"duration":7080}}]},"tags":["regression","reset_password"]},"source":"7e9477bf1d676706.json","parameterValues":[]}